storage/kafka/kafka.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	kafka "github.com/Shopify/sarama"
+	info "github.com/google/cadvisor/info/v1"
+	storage "github.com/google/cadvisor/storage"
+)
+
+func init() {
+	storage.RegisterStorageDriver("kafka", new)
+}
+
+type kafkaStorage struct {
+	producer    kafka.SyncProducer
+	topic       string
+	machineName string
+}
+
+type detailSpec struct {
+	Timestamp       int64                `json:"timestamp"`
+	MachineName     string               `json:"machine_name,omitempty"`
+	ContainerName   string               `json:"container_Name,omitempty"`
+	ContainerId     string               `json:"container_Id,omitempty"`
+	ContainerLabels map[string]string    `json:"container_labels,omitempty"`
+	ContainerStats  *info.ContainerStats `json:"container_stats,omitempty"`
+}
+
+func (driver *kafkaStorage) containerStatsAndDefaultValues(ref info.ContainerReference, stats *info.ContainerStats) *detailSpec {
+	timestamp := stats.Timestamp.UnixNano() / 1E3
```

Comment:
Microseconds is a bit of an unusual unit - is that a Kafka thing?

Also, for clarity, prefer:
```
timestamp := stats.Timestamp.UnixNano() / time.Microsecond
```